src/Chat/index.js

```diff
@@ -381,6 +368,10 @@ class Chat extends EventEmitter {
    * @return {Promise<undefined>}
    */
   whisper(user, message) {
+    if (chatUtils.isAnonymousUsername(this.options.username)) {
+      throw new Error('Not authenticated')
+    }
```

Comment:
Should we wrap this check in another method, like `Chat#isUserAuthorized`, and return a promise. Then, this promise can be added to the `Promise.all` of other methods?

While we are here, for consistency, `Chat#whisper` should return a resolved promise.

Reply:
Why not "asyncify" the method instead? By simply adding `async` before the method name?

Reply:
`async` / `await` is a new JavaScript paradigm; adding it would increase the _barrier_  to contribute. To keep the code simple and accessible to newer, less experienced developers, I would like to avoid `async` / `await`, at least for now.

Anyway, `Client#send` return a promise, so we could do something like:
```
this.isUserAuthenticated().then(() => this.send())
```